### PR TITLE
fix(utils): promote to next unit when rounding rolls over in formatCount/formatBytes

### PR DESCRIPTION
**Pull request by Osamaali313**
## What does this PR do?

`formatCount` and `formatBytes` (in `packages/utils`) choose a unit by dividing the value while it is `>= 1000` (or `1024`), then render it with `toFixed`. When the value sits just below the threshold, rounding pushes it back up to the threshold and produces nonsensical output:

- `formatCount(999999)` rendered `"1000K"` instead of `"1M"`
- `formatCount(999500)` rendered `"1000K"`
- `formatBytes(1024 * 1024 - 1)` rendered `"1024 KB"` instead of `"1 MB"`

After rounding, this promotes to the next unit when the rounded value reaches the threshold. All existing outputs are unchanged — only the rounding-boundary cases differ.

## Related issue (if applicable)

N/A — spotted while reading the formatting utilities.

## How was this tested?

Added regression tests in `packages/utils/src/format.test.ts` for both functions (the rollover cases plus the existing values). I also ran the edited `format.ts` directly and confirmed every existing case in the test file still produces the same output and that the boundary cases now render `"1M"` / `"1 MB"`.

## Checklist

- [ ] Lint, type-checking, and tests pass locally
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/)
- [ ] I have signed the CLA

### CODE REVIEW

**Comment by geclos** [comment]
@claude review

File: packages/utils/src/format.ts [hunk 1]
```diff
@@ -26,7 +26,14 @@ export function formatCount(count: number): string {
...
 
-  const decimal = value < 10 ? 1 : 0
+  let decimal = value < 10 ? 1 : 0
+  // Rounding can bump the value up to the next unit (e.g. 999_999 would render
+  // as "1000K"), so promote to the next unit when that happens.
+  if (Number(value.toFixed(decimal)) >= 1000 && unitIndex < COUNT_UNITS.length - 1) {
+    value /= 1000
+    unitIndex++
+    decimal = value < 10 ? 1 : 0
+  }
```
[the remark this answers is not in the document]
Comment (geclos) [reply packages/utils/src/format.ts:36]: @claude fix this

**Review by geclos — APPROVED** [review]
will fix the minor comments in another pr
